src/modules/previewpane/powerpreview/settings.cpp

```diff
@@ -60,12 +68,107 @@ namespace PowerPreviewSettings
 		}
 	}
 
+	LONG FileExplorerPreviewSettings::SetRegistryValue() const
+	{
+		HKEY hKey = HKEY_CURRENT_USER;
+		const REGSAM WRITE_PERMISSION = KEY_WRITE;
+		DWORD options = 0;
+		HKEY OpenResult;
+
+		LONG err = RegOpenKeyEx(hKey, this->GetSubKey(), options, WRITE_PERMISSION, &OpenResult); 
+
+		if (err == ERROR_SUCCESS)
+		{
+			err = RegSetValueExW(
+					OpenResult,
+					this->GetCLSID(),
+					0,
+					REG_SZ,
+					(LPBYTE)this->GetDisplayName().c_str(),
+					this->GetDisplayName().length() * sizeof(TCHAR));
+
+			if (err != ERROR_SUCCESS)
+			{
+				return err;
+			}
+		}
+		else
+		{
+			return err;
+		}
+        	RegCloseKey(OpenResult);
+	}
+
+	LONG FileExplorerPreviewSettings::RemvRegistryValue() const
+	{
+		HKEY hKey = HKEY_CURRENT_USER;
+		const REGSAM WRITE_PERMISSION = KEY_WRITE;
+		DWORD options = 0;
+		HKEY OpenResult;
+
+		LONG err = RegOpenKeyEx(hKey, this->GetSubKey(), options, WRITE_PERMISSION, &OpenResult); 
+		if (err == ERROR_SUCCESS)
+		{
+			err = RegDeleteKeyValueW(
+				OpenResult,
+				NULL,
+				this->GetCLSID());
+
+			if (err != ERROR_SUCCESS)
+			{
+				return err;
+			}
+		}
+		else
+		{
+			return err;
+		}
+        	RegCloseKey(OpenResult);
+	}
+
+	bool FileExplorerPreviewSettings::GetRegistryValue() const
+	{
+		HKEY OpenResult;
+		LONG err = RegOpenKeyEx(
+			HKEY_CURRENT_USER,
+			this->GetSubKey(), 
+			0, 
+			KEY_READ,
+			&OpenResult);
+
+		if (err != ERROR_SUCCESS)
```

Comment:
nit: this if/else has the reverse order compared to the other one

Reply:
I made the changes for consistency!